src/menu/menu.service.ts

```diff
@@ -1,18 +1,35 @@
 import { Injectable } from '@nestjs/common';
-import { MenuDto, Modification } from './dto/menu.dto';
+
 import { RestaurantService } from 'src/restaurant/restaurant.service';
-import { UserInfo } from 'src/common/decorators';
+import { AuthUser } from 'src/common/decorators';
 import { PrismaService } from 'src/prisma/prisma.service';
+import { CreateMenuItemDto, ModificationGroupDto } from './dto';
+import { ConfigService } from '@nestjs/config';
+import { PutObjectCommand, S3Client } from '@aws-sdk/client-s3';
+import { v4 as uuidv4 } from 'uuid';
+import { MenuItemImage } from '@prisma/client';
 
 @Injectable()
 export class MenuService {
-  menuId: number;
+  private readonly s3Client = new S3Client({
+    credentials: {
+      accessKeyId: this.configService.get('AWS_ACCESS_KEY'),
+      secretAccessKey: this.configService.get('AWS_SECRET_KEY'),
+    },
+    endpoint: this.configService.get('AWS_HOST'),
+    forcePathStyle: true,
+    region: this.configService.get('AWS_REGION'),
+  });
+  private menuItemImage: MenuItemImage;
```

Comment:
do not assign any result to class field.  

you can do so only if that class represents the object itself.  
for example: I have class Book.
then you can assign fields to it related to it (name, pages, etc.)

But when you have service class it cannot have field menuItemImage, that does not make sense and also will lead to error when service method is called from multiple places at the same time